tests/integration/test_charm.py

```diff
@@ -84,23 +86,158 @@ async def test_seldon_istio_relation(ops_test: OpsTest):
+
+    # verify that Seldon is in the target list
+    assert targets_result is not None
+    assert targets_result["status"] == "success"
+    discovered_labels = targets_result["data"]["activeTargets"][0]["discoveredLabels"]
+    assert discovered_labels["juju_application"] == "seldon-controller-manager"
+
+    # obtain alert rules from Prometheus
+    rules_url = f"http://{prometheus_url}:9090/api/v1/rules"
+    alert_rules_result = await fetch_url(rules_url)
+
+    # verify alerts are available in Prometheus
+    assert alert_rules_result is not None
+    assert alert_rules_result["status"] == "success"
+    rules = alert_rules_result["data"]["groups"][0]["rules"]
+
+    # load alert rules from the rules file
+    rules_file_alert_names = []
+    with open("src/prometheus_alert_rules/seldon_errors.rule") as f:
+        seldon_errors = yaml.safe_load(f.read())
+        alerts_list = seldon_errors["groups"][0]["rules"]
+        for alert in alerts_list:
+            rules_file_alert_names.append(alert["alert"])
+
+    # verify number of alerts is the same in Prometheus and in the rules file
+    assert len(rules) == len(rules_file_alert_names)
+
+    # verify that all Seldon alert rules are in the list and that alerts obtained from Prometheus
+    # match alerts in the rules file
+    for rule in rules:
+        assert rule["name"] in rules_file_alert_names
+
+    # The following integration test is optional (experimental) and might not be functioning
+    # correctly under some conditions due to its reliance on timing of K8S deployments, timing of
+    # Prometheus scraping, and rate calculations for alerts.
+    # In addition, Seldon Core Operator has one relatively easily triggered alert
+    # (SeldonReconcileError) that can be simulated.
+
+    # Test 2: Simulate propagattion of SeldonReconcileError alert by deleting deployment.
+
+    test_alert_name = "SeldonReconcileError"
+
+    # verify that alert SeldonReconcileError is inactive
+    seldon_reconcile_error_alert = next(
+        (rule for rule in rules if rule["name"] == test_alert_name)
+    )
+    assert seldon_reconcile_error_alert["state"] == "inactive"
+
+    # simulate scenario where alert will fire
+    # create SeldonDeployment
+    seldon_deployment = create_namespaced_resource(
+        group="machinelearning.seldon.io",
+        version="v1",
+        kind="seldondeployment",
+        plural="seldondeployments",
+        verbs=None,
+    )
+    with open("examples/serve-simple-v1.yaml") as f:
+        sdep = seldon_deployment(yaml.safe_load(f.read()))
+        sdep["metadata"]["name"] = "seldon-model-1"
+        client.create(sdep, namespace=namespace)
+    assert_available(client, seldon_deployment, "seldon-model-1", namespace)
+
+    # remove deployment that was created by Seldon, reconcile alert will fire
+    client.delete(Deployment, name="seldon-model-1-example-0-classifier", namespace=namespace)
```

Comment:
I think this delete call must refer to the `SeldonDeployment` instead, you can use the resource class from the previous lines `seldon_deployment`

Reply:
Not quite. This is exactly the trick I am playing on Seldon to trigger the alert. SeldonDeployment still exists, but classifier deployment `seldon-model-1-example-0-classifier` is removed. This causes reconciliation, which in turn triggers reconcile error alert.
If I remove SeldonDeployment, no alerts will be triggered.

Reply:
Makes sense.